src/resolve/resolved-dns-rr.c

```diff
@@ -1043,8 +1042,10 @@ const char *dns_resource_record_to_string(DnsResourceRecord *rr) {
                 r = base64_append(&s, r,
                                   rr->rrsig.signature, rr->rrsig.signature_size,
                                   8, columns());
-                if (r < 0)
+                if (r < 0) {
+                        free(s);
```

Comment:
wouldn't it be safer to use autoclean for `s` as well?

Reply:
Possibly. I was hesitating which variant of the patch to submit; in the end I chose this one just because it's slightly shorter. But I don't really like either version. I even thought about introducing a wrapper around `asprintf()` that guarantees that the output string is not touched on error. With that, a single auto-cleaned variable would be sufficient to build the output string for all the cases...

Reply:
I'd prefer this would use cleanup stuff too, just like the stuff you fixed above, i.e. just use `t` for the intermediary stuff, we have it anyway